### ISSUE TITLE
Electron app not starting

### ISSUE THREAD

**Issue by J-Cake** [canonical/ubuntu-frame#176]
I would give more info, but I simply don't have any...

I want to run Obsidian on Frame but am struggling to get very far. Ubuntu Frame starts and shows the typical greydient (lool) but when running Obsidian, both via the CLI directly and via `frame-it`, I'm unsuccessful. Below is the output from `WAYLAND-DISPLAY=wayland-99 sudo -E obsidian --enable-features=UseOzonePlatform --ozone-platform=wayland`, but I've tried all sorts of combinations, read through basically all forums etc. Hoping for some help.

Thanks

```
jcake@jcake-edu:~$ sudo frame-it obsidian --enable-features=UseOzonePlatform --ozone-platform=wayland 
+ [ /snap/ubuntu-frame/8587/bin/run-frame = --help ]
+ mkdir -p /root/snap/ubuntu-frame/8587/.config/
+ grep -vE (^vt=|^console-provider=vt) /var/snap/ubuntu-frame/8587/frame.config
+ rm -f /root/snap/ubuntu-frame/8587/.config/frame.display
+ cp /var/snap/ubuntu-frame/8587/frame.display /root/snap/ubuntu-frame/8587/.config/
cp: cannot stat '/var/snap/ubuntu-frame/8587/frame.display': No such file or directory
+ true
+ exec /snap/ubuntu-frame/8587/bin/run-frame /snap/ubuntu-frame/8587/bin/graphics-core22-wrapper /snap/ubuntu-frame/8587/usr/local/bin/frame
+ [ /snap/ubuntu-frame/8587/bin/graphics-core22-wrapper = --help ]
+ dirname /run/user/0/snap.ubuntu-frame
+ XDG_RUNTIME_DIR=/run/user/0
+ export XDG_RUNTIME_DIR
+ mkdir -p /run/user/0 -m 700
+ [ -n wayland-0 ]
+ [ -e /run/user/0/wayland-0 ]
+ exec /snap/ubuntu-frame/8587/bin/graphics-core22-wrapper /snap/ubuntu-frame/8587/usr/local/bin/frame
[2024-04-13 21:30:54.157165] <information> mirserver: Starting
[2024-04-13 21:30:54.160464] < - debug - > mirserver: Not using logind for session management: Failed to find active session
[2024-04-13 21:30:54.160654] < - debug - > mirserver: Using Linux VT subsystem for session management
[2024-04-13 21:30:54.160740] <information> mircommon: Loading modules from: /snap/ubuntu-frame/8587/usr/lib/x86_64-linux-gnu/mir/server-platform
[2024-04-13 21:30:54.160766] <information> mircommon: Loading module: /snap/ubuntu-frame/8587/usr/lib/x86_64-linux-gnu/mir/server-platform/graphics-eglstream-kms.so.21
[2024-04-13 21:30:54.160776] <information> mircommon: Loading module: /snap/ubuntu-frame/8587/usr/lib/x86_64-linux-gnu/mir/server-platform/graphics-gbm-kms.so.21
[2024-04-13 21:30:54.160781] <information> mircommon: Loading module: /snap/ubuntu-frame/8587/usr/lib/x86_64-linux-gnu/mir/server-platform/renderer-egl-generic.so.21
[2024-04-13 21:30:54.160786] <information> mircommon: Loading module: /snap/ubuntu-frame/8587/usr/lib/x86_64-linux-gnu/mir/server-platform/server-x11.so.21
[2024-04-13 21:30:54.160790] <information> mircommon: Loading module: /snap/ubuntu-frame/8587/usr/lib/x86_64-linux-gnu/mir/server-platform/input-evdev.so.8
[2024-04-13 21:30:54.161175] <information> mirserver: Found display driver: mir:eglstream-kms (version 2.16.3)
[2024-04-13 21:30:54.161663] < - debug - > eglstream: Found EGLDeviceEXT with device extensions: EGL_MESA_device_software EGL_EXT_device_drm_render_node
[2024-04-13 21:30:54.161689] <information> eglstream: Failed to query DRM node for EGLDevice: Failed to determine DRM device node path from EGLDevice: EGL_BAD_PARAMETER (0x300c)
[2024-04-13 21:30:54.161693] < - debug - > eglstream: EGLDeviceEXTs found, but none are suitable for Mir
[2024-04-13 21:30:54.161697] <information> mirserver: (Unsupported by system environment)
[2024-04-13 21:30:54.161702] <information> mirserver: Found display driver: mir:gbm-kms (version 2.16.3)
[2024-04-13 21:30:54.161867] < - debug - > gbm-kms: Quirks: checking device with devnode: /dev/dri/card0, driver bochs-drm
pci id for fd 12: 1234:1111, driver (null)
MESA-LOADER: failed to open bochs-drm: /snap/ubuntu-frame/8587/graphics/usr/lib/i386-linux-gnu/dri//bochs-drm_dri.so: cannot open shared object file: No such file or directory (search paths /snap/ubuntu-frame/8587/graphics/usr/lib/x86_64-linux-gnu/dri/:/snap/ubuntu-frame/8587/graphics/usr/lib/i386-linux-gnu/dri/, suffix _dri)
pci id for fd 11: 1234:1111, driver (null)
kmsro: driver missing
Failed to create /root/.cache for shader cache (Permission denied)---disabling.
[2024-04-13 21:30:54.175964] < - debug - > gbm-kms: Quirks: checking device with devnode: (null), driver 
[2024-04-13 21:30:54.176044] <information> mirserver: Driver supports:
[2024-04-13 21:30:54.176053] <information> mirserver:   /devices/pci0000:00/0000:00:02.0/drm/card0 (priority 256)
[2024-04-13 21:30:54.176062] <information> mirserver: Found display driver: mir:egl-generic (version 2.16.3)
[2024-04-13 21:30:54.176087] <information> mirserver: Found display driver: mir:x11 (version 2.16.3)
[2024-04-13 21:30:54.176101] <information> mirserver: (Unsupported by system environment)
[2024-04-13 21:30:54.176117] <information> mirserver: Selected display driver: mir:gbm-kms (version 2.16.3) for device ((null): /dev/dri/card0)
pci id for fd 12: 1234:1111, driver (null)
MESA-LOADER: failed to open bochs-drm: /snap/ubuntu-frame/8587/graphics/usr/lib/i386-linux-gnu/dri//bochs-drm_dri.so: cannot open shared object file: No such file or directory (search paths /snap/ubuntu-frame/8587/graphics/usr/lib/x86_64-linux-gnu/dri/:/snap/ubuntu-frame/8587/graphics/usr/lib/i386-linux-gnu/dri/, suffix _dri)
pci id for fd 11: 1234:1111, driver (null)
kmsro: driver missing
[2024-04-13 21:30:54.178632] <information> mircommon: Loading modules from: /snap/ubuntu-frame/8587/usr/lib/x86_64-linux-gnu/mir/server-platform
[2024-04-13 21:30:54.178657] <information> mircommon: Loading module: /snap/ubuntu-frame/8587/usr/lib/x86_64-linux-gnu/mir/server-platform/graphics-eglstream-kms.so.21
[2024-04-13 21:30:54.178664] <information> mircommon: Loading module: /snap/ubuntu-frame/8587/usr/lib/x86_64-linux-gnu/mir/server-platform/graphics-gbm-kms.so.21
[2024-04-13 21:30:54.178669] <information> mircommon: Loading module: /snap/ubuntu-frame/8587/usr/lib/x86_64-linux-gnu/mir/server-platform/renderer-egl-generic.so.21
[2024-04-13 21:30:54.178674] <information> mircommon: Loading module: /snap/ubuntu-frame/8587/usr/lib/x86_64-linux-gnu/mir/server-platform/server-x11.so.21
[2024-04-13 21:30:54.178678] <information> mircommon: Loading module: /snap/ubuntu-frame/8587/usr/lib/x86_64-linux-gnu/mir/server-platform/input-evdev.so.8
[2024-04-13 21:30:54.179053] <information> mirserver: Found rendering driver: mir:eglstream-kms (version 2.16.3)
[2024-04-13 21:30:54.179184] < - debug - > eglstream: Failed to find kernel device for EGLDevice: Failed to determine DRM device node path from EGLDevice: EGL_BAD_PARAMETER (0x300c)
[2024-04-13 21:30:54.179189] < - debug - > eglstream: EGLDeviceEXTs found, but none are suitable for Mir
[2024-04-13 21:30:54.179192] <information> mirserver: (Unsupported by system environment)
[2024-04-13 21:30:54.179196] <information> mirserver: Found rendering driver: mir:gbm-kms (version 2.16.3)
[2024-04-13 21:30:54.179202] < - debug - > gbm-kms: GBM-capable display found
[2024-04-13 21:30:54.179361] < - debug - > gbm-kms: Quirks: checking device with devnode: /dev/dri/card0, driver bochs-drm
pci id for fd 13: 1234:1111, driver (null)
MESA-LOADER: failed to open bochs-drm: /snap/ubuntu-frame/8587/graphics/usr/lib/i386-linux-gnu/dri//bochs-drm_dri.so: cannot open shared object file: No such file or directory (search paths /snap/ubuntu-frame/8587/graphics/usr/lib/x86_64-linux-gnu/dri/:/snap/ubuntu-frame/8587/graphics/usr/lib/i386-linux-gnu/dri/, suffix _dri)
pci id for fd 14: 1234:1111, driver (null)
kmsro: driver missing
Failed to create /root/.cache for shader cache (Permission denied)---disabling.
[2024-04-13 21:30:54.181740] <information> gbm-kms: Detected software renderer: llvmpipe (LLVM 15.0.7, 256 bits)
[2024-04-13 21:30:54.190175] < - debug - > gbm-kms: Quirks: checking device with devnode: (null), driver 
[2024-04-13 21:30:54.190191] <information> mirserver: Driver supports:
[2024-04-13 21:30:54.190196] <information> mirserver:   /devices/pci0000:00/0000:00:02.0/drm/card0 (priority 0)
[2024-04-13 21:30:54.190202] <information> mirserver: Found rendering driver: mir:egl-generic (version 2.16.3)
libEGL warning: pci id for fd 13: 1234:1111, driver (null)

[2024-04-13 21:30:54.190972] <information> mirserver: Driver supports:
[2024-04-13 21:30:54.190978] <information> mirserver:   System (priority 128)
[2024-04-13 21:30:54.190986] <information> mirserver: Found rendering driver: mir:x11 (version 2.16.3)
[2024-04-13 21:30:54.191009] <information> mirserver: Selected rendering driver: mir:egl-generic (version 2.16.3) for platform
libEGL warning: pci id for fd 13: 1234:1111, driver (null)

Failed to create /root/.cache for shader cache (Permission denied)---disabling.
[2024-04-13 21:30:54.194787] <information> platform-generic-egl: Cannot enable linux-dmabuf import support: EGL_EXT_image_dma_buf_import_modifiers not supported
[2024-04-13 21:30:54.194820] < - debug - > platform-generic-egl: Detailed error: : ./src/platform/graphics/egl_extensions.cpp(225): Throw in function mir::graphics::EGLExtensions::EXTImageDmaBufImportModifiers::EXTImageDmaBufImportModifiers(EGLDisplay)
Dynamic exception type: boost::wrapexcept<std::runtime_error>
std::exception::what: EGL_EXT_image_dma_buf_import_modifiers not supported

[2024-04-13 21:30:54.195330] < -warning- > gbm-kms: Unable to determine the current display mode.
[2024-04-13 21:30:54.195389] <information> gbm-kms: DRM device details:
[2024-04-13 21:30:54.195399] <information> gbm-kms: /dev/dri/card0: using driver bochs-drm [bochs dispi vga interface (qemu stdvga)] (version: 1.0.0 driver date: 20130925)
[2024-04-13 21:30:54.195506] <information> gbm-kms:     Output: Virtual-1 (connected)
[2024-04-13 21:30:54.195514] <information> gbm-kms:             Mode: 1280×800@74.99
[2024-04-13 21:30:54.195519] <information> gbm-kms:             Mode: 1920×1080@60.00
[2024-04-13 21:30:54.195524] <information> gbm-kms:             Mode: 1920×1080@50.00
[2024-04-13 21:30:54.195529] <information> gbm-kms:             Mode: 1600×1200@60.00
[2024-04-13 21:30:54.195534] <information> gbm-kms:             Mode: 1680×1050@59.95
[2024-04-13 21:30:54.195539] <information> gbm-kms:             Mode: 1400×1050@59.97
[2024-04-13 21:30:54.195544] <information> gbm-kms:             Mode: 1280×1024@60.01
[2024-04-13 21:30:54.195548] <information> gbm-kms:             Mode: 1440×900@59.88
[2024-04-13 21:30:54.195553] <information> gbm-kms:             Mode: 1280×960@60.00
[2024-04-13 21:30:54.195558] <information> gbm-kms:             Mode: 1360×768@60.01
[2024-04-13 21:30:54.195563] <information> gbm-kms:             Mode: 1280×768@59.87
[2024-04-13 21:30:54.195568] <information> gbm-kms:             Mode: 1024×768@60.00
[2024-04-13 21:30:54.195573] <information> gbm-kms:             Mode: 800×600@60.31
[2024-04-13 21:30:54.195577] <information> gbm-kms:             Mode: 640×480@60.00
[2024-04-13 21:30:54.195582] <information> gbm-kms:             Mode: 640×480@59.94
[2024-04-13 21:30:54.195591] < - debug - > miral: Display config using layout strategy: 'default'
[2024-04-13 21:30:54.195615] <information> miral: Display config:
8>< ---------------------------------------------------
layouts:
  default:                         # the current layout
    cards:
    # a list of cards (currently matched by card-id)

    - card-id: 0
      Virtual-1:
        # This output supports the following modes: 1280x800@75.0, 1920x1080@60.0,
        # 1920x1080@50.0, 1600x1200@60.0, 1680x1050@60.0, 1400x1050@60.0, 1280x1024@60.0,
        # 1440x900@59.9, 1280x960@60.0, 1360x768@60.0, 1280x768@59.9, 1024x768@60.0,
        # 800x600@60.3, 640x480@60.0, 640x480@59.9
        #
        # Uncomment the following to enforce the selected configuration.
        # Or amend as desired.
        #
        state: enabled  # {enabled, disabled}, defaults to enabled
        mode: 1280x800@75.0     # Defaults to preferred mode
        position: [0, 0]        # Defaults to [0, 0]
        orientation: normal     # {normal, left, right, inverted}, defaults to normal
        scale: 1
        group: 0        # Outputs with the same non-zero value are treated as a single display
8>< ---------------------------------------------------
[2024-04-13 21:30:54.195843] < -warning- > gbm-kms: drmModeCrtcSetGamma failed: Function not implemented
[2024-04-13 21:30:54.195860] < - debug - > miral: Display config using layout strategy: 'default'
[2024-04-13 21:30:54.195873] <information> miral: Display config:
8>< ---------------------------------------------------
layouts:
  default:                         # the current layout
    cards:
    # a list of cards (currently matched by card-id)

    - card-id: 0
      Virtual-1:
        # This output supports the following modes: 1280x800@75.0, 1920x1080@60.0,
        # 1920x1080@50.0, 1600x1200@60.0, 1680x1050@60.0, 1400x1050@60.0, 1280x1024@60.0,
        # 1440x900@59.9, 1280x960@60.0, 1360x768@60.0, 1280x768@59.9, 1024x768@60.0,
        # 800x600@60.3, 640x480@60.0, 640x480@59.9
        #
        # Uncomment the following to enforce the selected configuration.
        # Or amend as desired.
        #
        state: enabled  # {enabled, disabled}, defaults to enabled
        mode: 1280x800@75.0     # Defaults to preferred mode
        position: [0, 0]        # Defaults to [0, 0]
        orientation: normal     # {normal, left, right, inverted}, defaults to normal
        scale: 1
        group: 0        # Outputs with the same non-zero value are treated as a single display
8>< ---------------------------------------------------
[2024-04-13 21:30:54.195929] < - debug - > miral: Wrote display configuration template: /root/snap/ubuntu-frame/8587/.config/frame.display
xkbcommon: ERROR: couldn't find a Compose file for locale "C.UTF-8" (mapped to "C.UTF-8")
[2024-04-13 21:30:54.198634] <information> mirserver: Using software cursor
[2024-04-13 21:30:54.198735] < -warning- > gbm-kms-buffer-allocator: Failed to bind EGL Display to Wayland display, falling back to software buffers: ./src/platform/graphics/egl_extensions.cpp(111): Throw in function mir::graphics::EGLExtensions::WaylandExtensions::WaylandExtensions(EGLDisplay)
Dynamic exception type: boost::wrapexcept<std::runtime_error>
std::exception::what: EGL display doesn't support EGL_WL_bind_wayland_display

[2024-04-13 21:30:54.198833] <information> mircommon: Loading modules from: /snap/ubuntu-frame/8587/usr/lib/x86_64-linux-gnu/mir/server-platform
[2024-04-13 21:30:54.198854] <information> mircommon: Loading module: /snap/ubuntu-frame/8587/usr/lib/x86_64-linux-gnu/mir/server-platform/graphics-eglstream-kms.so.21
[2024-04-13 21:30:54.198866] <information> mircommon: Loading module: /snap/ubuntu-frame/8587/usr/lib/x86_64-linux-gnu/mir/server-platform/graphics-gbm-kms.so.21
[2024-04-13 21:30:54.198873] <information> mircommon: Loading module: /snap/ubuntu-frame/8587/usr/lib/x86_64-linux-gnu/mir/server-platform/renderer-egl-generic.so.21
[2024-04-13 21:30:54.198881] <information> mircommon: Loading module: /snap/ubuntu-frame/8587/usr/lib/x86_64-linux-gnu/mir/server-platform/server-x11.so.21
[2024-04-13 21:30:54.198888] <information> mircommon: Loading module: /snap/ubuntu-frame/8587/usr/lib/x86_64-linux-gnu/mir/server-platform/input-evdev.so.8
[2024-04-13 21:30:54.199297] <information> mirserver: Selected input driver: mir:evdev-input (version: 2.16.3)
[2024-04-13 21:30:54.199370] <information> mirserver: Mir version 2.16.3
[2024-04-13 21:30:54.206135] <information> GLRenderer: EGL vendor: Mesa Project
[2024-04-13 21:30:54.206153] <information> GLRenderer: EGL version: 1.5
[2024-04-13 21:30:54.206156] <information> GLRenderer: EGL client APIs: OpenGL OpenGL_ES 
[2024-04-13 21:30:54.206159] <information> GLRenderer: EGL extensions: EGL_EXT_create_context_robustness EGL_KHR_cl_event2 EGL_KHR_config_attribs EGL_KHR_context_flush_control EGL_KHR_create_context EGL_KHR_create_context_no_error EGL_KHR_fence_sync EGL_KHR_get_all_proc_addresses EGL_KHR_gl_colorspace EGL_KHR_gl_renderbuffer_image EGL_KHR_gl_texture_2D_image EGL_KHR_gl_texture_3D_image EGL_KHR_gl_texture_cubemap_image EGL_KHR_image_base EGL_KHR_no_config_context EGL_KHR_reusable_sync EGL_KHR_surfaceless_context EGL_EXT_pixel_format_float EGL_KHR_wait_sync EGL_MESA_configless_context EGL_MESA_drm_image EGL_MESA_query_driver 
[2024-04-13 21:30:54.206167] <information> GLRenderer: GL vendor: Mesa
[2024-04-13 21:30:54.206172] <information> GLRenderer: GL renderer: llvmpipe (LLVM 15.0.7, 256 bits)
[2024-04-13 21:30:54.206175] <information> GLRenderer: GL version: OpenGL ES 3.2 Mesa 23.0.4-0ubuntu1~22.04.1
[2024-04-13 21:30:54.206179] <information> GLRenderer: GLSL version: OpenGL ES GLSL ES 3.20
[2024-04-13 21:30:54.206209] <information> GLRenderer: GL extensions: GL_EXT_blend_minmax GL_EXT_multi_draw_arrays GL_EXT_texture_filter_anisotropic GL_EXT_texture_compression_s3tc GL_EXT_texture_compression_dxt1 GL_EXT_texture_compression_rgtc GL_EXT_texture_format_BGRA8888 GL_OES_compressed_ETC1_RGB8_texture GL_OES_depth24 GL_OES_element_index_uint GL_OES_fbo_render_mipmap GL_OES_mapbuffer GL_OES_rgb8_rgba8 GL_OES_standard_derivatives GL_OES_stencil8 GL_OES_texture_3D GL_OES_texture_float GL_OES_texture_float_linear GL_OES_texture_half_float GL_OES_texture_half_float_linear GL_OES_texture_npot GL_OES_vertex_half_float GL_EXT_draw_instanced GL_EXT_texture_sRGB_decode GL_OES_EGL_image GL_OES_depth_texture GL_OES_packed_depth_stencil GL_EXT_texture_type_2_10_10_10_REV GL_NV_conditional_render GL_OES_get_program_binary GL_APPLE_texture_max_level GL_EXT_discard_framebuffer GL_EXT_read_format_bgra GL_NV_pack_subimage GL_EXT_frag_depth GL_NV_fbo_color_attachments GL_OES_EGL_image_external GL_OES_EGL_sync GL_OES_vertex_array_object GL_OES_viewport_array GL_ANGLE_pack_reverse_row_order GL_ANGLE_texture_compression_dxt3 GL_ANGLE_texture_compression_dxt5 GL_EXT_occlusion_query_boolean GL_EXT_robustness GL_EXT_texture_rg GL_EXT_unpack_subimage GL_NV_draw_buffers GL_NV_read_buffer GL_NV_read_depth GL_NV_read_depth_stencil GL_NV_read_stencil GL_EXT_draw_buffers GL_EXT_map_buffer_range GL_KHR_debug GL_KHR_robustness GL_KHR_texture_compression_astc_ldr GL_NV_pixel_buffer_object GL_OES_depth_texture_cube_map GL_OES_required_internalformat GL_OES_surfaceless_context GL_EXT_color_buffer_float GL_EXT_debug_label GL_EXT_sRGB_write_control GL_EXT_separate_shader_objects GL_EXT_shader_framebuffer_fetch GL_EXT_shader_group_vote GL_EXT_shader_implicit_conversions GL_EXT_shader_integer_mix GL_EXT_tessellation_point_size GL_EXT_tessellation_shader GL_ANDROID_extension_pack_es31a GL_ARM_shader_framebuffer_fetch_depth_stencil GL_EXT_base_instance GL_EXT_compressed_ETC1_RGB8_sub_texture GL_EXT_copy_image GL_EXT_draw_buffers_indexed GL_EXT_draw_elements_base_vertex GL_EXT_gpu_shader5 GL_EXT_polygon_offset_clamp GL_EXT_primitive_bounding_box GL_EXT_render_snorm GL_EXT_shader_io_blocks GL_EXT_texture_border_clamp GL_EXT_texture_buffer GL_EXT_texture_cube_map_array GL_EXT_texture_norm16 GL_EXT_texture_view GL_KHR_blend_equation_advanced GL_KHR_blend_equation_advanced_coherent GL_KHR_context_flush_control GL_KHR_robust_buffer_access_behavior GL_NV_image_formats GL_NV_shader_noperspective_interpolation GL_OES_copy_image GL_OES_draw_buffers_indexed GL_OES_draw_elements_base_vertex GL_OES_gpu_shader5 GL_OES_primitive_bounding_box GL_OES_sample_shading GL_OES_sample_variables GL_OES_shader_io_blocks GL_OES_shader_multisample_interpolation GL_OES_tessellation_point_size GL_OES_tessellation_shader GL_OES_texture_border_clamp GL_OES_texture_buffer GL_OES_texture_cube_map_array GL_OES_texture_stencil8 GL_OES_texture_storage_multisample_2d_array GL_OES_texture_view GL_EXT_blend_func_extended GL_EXT_buffer_storage GL_EXT_float_blend GL_EXT_geometry_point_size GL_EXT_geometry_shader GL_EXT_texture_filter_minmax GL_EXT_texture_sRGB_R8 GL_EXT_texture_sRGB_RG8 GL_KHR_no_error GL_KHR_texture_compression_astc_sliced_3d GL_OES_EGL_image_external_essl3 GL_OES_geometry_point_size GL_OES_geometry_shader GL_OES_shader_image_atomic GL_EXT_clear_texture GL_EXT_clip_cull_distance GL_EXT_disjoint_timer_query GL_EXT_texture_compression_s3tc_srgb GL_MESA_shader_integer_functions GL_EXT_clip_control GL_EXT_color_buffer_half_float GL_EXT_memory_object GL_EXT_memory_object_fd GL_EXT_texture_compression_bptc GL_EXT_texture_mirror_clamp_to_edge GL_KHR_parallel_shader_compile GL_EXT_EGL_image_storage GL_EXT_shader_framebuffer_fetch_non_coherent GL_EXT_texture_shadow_lod GL_MESA_framebuffer_flip_y GL_EXT_depth_clamp GL_EXT_texture_query_lod GL_MESA_bgra 
[2024-04-13 21:30:54.206233] <information> GLRenderer: GL max texture size = 16384
[2024-04-13 21:30:54.206754] <information> GLRenderer: GL framebuffer bits: RGBA=8888, depth=0, stencil=0
[2024-04-13 21:30:54.207941] <information> mirserver: Initial display configuration:
[2024-04-13 21:30:54.207951] <information> mirserver: * Output 1: Virtual connected, used
[2024-04-13 21:30:54.207955] <information> mirserver: . |_ EDID monitor name: QEMU Monitor
[2024-04-13 21:30:54.207960] <information> mirserver: . |_ EDID manufacturer: RHT
[2024-04-13 21:30:54.207965] <information> mirserver: . |_ EDID product code: 4660
[2024-04-13 21:30:54.207972] <information> mirserver: . |_ Physical size 14.9" 320x200mm
[2024-04-13 21:30:54.207977] <information> mirserver: . |_ Power is on
[2024-04-13 21:30:54.207982] <information> mirserver: . |_ Current mode 1280x800 74.99Hz
[2024-04-13 21:30:54.207986] <information> mirserver: . |_ Preferred mode 1280x800 74.99Hz
[2024-04-13 21:30:54.207991] <information> mirserver: . |_ Orientation normal
[2024-04-13 21:30:54.207996] <information> mirserver: . |_ Logical size 1280x800
[2024-04-13 21:30:54.208001] <information> mirserver: . |_ Logical position +0+0
[2024-04-13 21:30:54.208005] <information> mirserver: . |_ Scaling factor: 1.00
[2024-04-13 21:30:54.208022] < -warning- > miral: Attempt to read custom output attribute (surface-title) that wasn't added
[2024-04-13 21:30:54.208033] < -warning- > miral: Attempt to read custom output attribute (snap-name) that wasn't added
[2024-04-13 21:31:01.160244] <information> evdev-input: Added /dev/input/event0
[2024-04-13 21:31:01.163231] <information> evdev-input: Added /dev/input/event1
[2024-04-13 21:31:01.164826] <information> evdev-input: Added /dev/input/event3
[2024-04-13 21:31:01.164874] <information> evdev-input: Added /dev/input/event2
```

**Comment by AlanGriffiths** [comment]
> jcake@jcake-edu:~$ sudo frame-it obsidian --enable-features=UseOzonePlatform --ozone-platform=wayland

Why are you running `frame-it` as root? What happens with `frame-it obsidian --enable-features=UseOzonePlatform --ozone-platform=wayland`?

> I've tried all sorts of combinations, read through basically all forums etc. Hoping for some help

This is probably what you need:

https://mir-server.io/docs/packaging-an-electron-application-as-an-iot-gui

Closing issue as this is not a problem with Ubuntu Frame

**Comment by J-Cake** [comment]
I'm running `frame-it` as root to make sure it isn't that. As I mentioned, I've tried basically everything. I've indeed been following that article too, and have had very little luck.

**Comment by AlanGriffiths** [comment]
> I'm running `frame-it` as root to make sure it isn't that. As I mentioned, I've tried basically everything. I've indeed been following that article too, and have had very little luck.

Running as root _will cause problems_: the log from doing so is unlikely to provide useful information on why Obsidian doesn't do what you expect. 

> Ubuntu Frame starts and shows the typical greydient

This shows that Frame starts and implies that any problem is not with Frame.

> when running Obsidian, both via the CLI directly

So far as I can see Obsidian is not in the Ubuntu archive, so any problem is likely to be with the way it is built, packaged or installed.

Errors from Obsidian (ideally run via the CLI) might indicate why it doesn't start

**Comment by J-Cake** [comment]
I see. When I'm at my computer next, I'll run it again as standard. Obsidian is a available as both a snap and directly in the package repos. I've tried both. I also tried it with VSCode packaged in the same (or at least similar) way and had the same results. 

I also forgot to mention that running frame-it on my desktop the way the article describes did work, but only if the electron app is started with the necessary wayland parameters
